### PR TITLE
speaker sample

### PR DESCRIPTION
**Pull request by funkychicken493**
Shows an example of how to add a custom redstone-powerable block.

### CODE REVIEW

File: external/BlockArchitect/custom_blocks/samples/speaker.dsc [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        custom_model_data: 3
+    flags:
+        custom_block: true
+
+custom_speaker_activate:
+    type: world
+    debug: false
+    events:
+        on custom event id:blockarchitect_block_switched:
+        - if <context.blockarchitect_id> != custom_speaker:
```
Comment (Hydroxycobalamin) [inline external/BlockArchitect/custom_blocks/samples/speaker.dsc:17]: You can make use of the data:<key>:<value> switch of the customevent event.

File: external/BlockArchitect/custom_blocks/samples/speaker.dsc [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+            - stop
+        - if <context.location.switched>:
```
Comment (Hydroxycobalamin) [inline external/BlockArchitect/custom_blocks/samples/speaker.dsc:19]: <context.switched>

File: external/BlockArchitect/custom_blocks/samples/speaker.dsc [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+            - playsound <context.location> sound:entity_cat_ambient
+            - playeffect at:<context.location.center.above[0.55]> effect:vibration special_data:1s|<context.location.center.above[0.55]>|<context.location.center.above[1.5]> offset:0.01 quantity:50
+        - switch <context.location> off no_physics
```
Comment (Hydroxycobalamin) [inline external/BlockArchitect/custom_blocks/samples/speaker.dsc:22]: `state:`

**Comment by funkychicken493** [comment]
@Hydroxycobalamin your suggestions have been implemented.

**Comment by funkychicken493** [comment]
Issue with PR discovered: placing a block adjacent to the lamp whilst it is powered (but the real block isn't lit) will result in it falsely triggering. Attempts to fix this have been fruitless.

**Pull request merged**